py-deploy/tldeploy/core.py

```diff
@@ -37,29 +42,65 @@ def get_contract_interface(contract_name):
     return contracts[contract_name]
 
 
-def deploy(contract_name, web3, *args):
+def deploy(
+    contract_name,
+    *,
+    web3: Web3,
+    transaction_options: Dict = {},
```

Comment:
afaik it is bad to have mutuables as default args in functions, because the default is only instantiated once and if you ever modify it for any reason, then every other call will use a different default.

I think you should do:
`transacation_options = None` and then `if transaction_options is None: transaction_option = {}`

Reply:
Okay, interesting!
I think I can set it to `None` everywhere and don't care further. The `deploy-tools` will set it in the end to `{}` if it is `None`. So I don't have to put this if-clause everywhere. :+1:

Reply:
Damn. Our `increase_transaction_options_nonce()` function is stupid. :cry:

Reply:
Why is it stupid?

Reply:
`if "nonce" in transaction_options:`.
Every part of the code is aware that it can be `None`, except this function. xD

Reply:
We can change the behavior of the `contract-deploy-tools` in future. At the moment it is a too small change with too much work to integrate a new version.
Therefore I fixed it as you suggested it. Looking forward to your approve. ;)